### PR TITLE
Add "Notice" Logging Level

### PR DESCRIPTION
**Pull request by Byrdman32**
#### Summary
This PR adds a new "notice" logging level to the Quill logging library. The notice level is intended to capture events that are significant, but not necessarily errors, warnings, or informational logs. It fits between the `INFO` and `WARNING` levels and can be used for logging important runtime events that should be brought to the user's attention without signaling an issue with the program's behavior.

#### Changes
- Introduced a new `NOTICE` level to the logging hierarchy.
- Defined a corresponding integer value for the `NOTICE` level that fits between `INFO` and `WARNING`.
- Updated the logger to support logging with `NOTICE` level.
- Added tests to verify correct behavior of the `NOTICE` level.
- Updated documentation to reflect the addition of the `NOTICE` level.

#### Example Usage
```cpp
LOG_NOTICE(logger, "This is a notice message.");
```

#### Motivation
The addition of the `NOTICE` level provides developers with a more granular approach to logging. It helps distinguish between routine information and noteworthy events, without the higher severity of warnings. This can be particularly useful for logging important but expected events, system state changes, or key milestones in the application's lifecycle.

#### Tests
- Added unit tests for logging at the `NOTICE` level.
- Verified that the `NOTICE` level works seamlessly across all sinks and formats as expected.

#### Impact
- No breaking changes.
- Backward compatibility maintained.
- New feature available for projects that need a more nuanced logging level between `INFO` and `WARNING`.

#### Checklist
- [x] Added a new `NOTICE` logging level.
- [x] Updated relevant tests.
- [x] Updated documentation.
- [x] Verified that the changes are backward compatible.

### CODE REVIEW

File: include/quill/LogMacros.h [hunk 1]
```diff
@@ -580,6 +581,51 @@
...
 
+#if QUILL_COMPILE_ACTIVE_LOG_LEVEL <= QUILL_COMPILE_ACTIVE_LOG_LEVEL_NOTICE
+  #define QUILL_LOG_NOTICE(logger, fmt, ...)                                                         \
+    QUILL_LOGGER_CALL(QUILL_LIKELY, logger, quill::LogLevel::Notice, fmt, ##__VA_ARGS__)
+
+  #define QUILL_LOG_NOTICE_LIMIT(min_interval, logger, fmt, ...)                                     \
+    QUILL_LOGGER_CALL_LIMIT(min_interval, QUILL_LIKELY, logger, quill::LogLevel::Notice, fmt, ##__VA_ARGS__)
+
+  #define QUILL_LOG_NOTICE_WITH_TAGS(logger, tags, fmt, ...)                                         \
+    QUILL_LOGGER_CALL_WITH_TAGS(QUILL_LIKELY, logger, quill::LogLevel::Notice, tags, fmt, ##__VA_ARGS__)
```
Comment (odygrd) [inline include/quill/LogMacros.h:592]: This has been replaced on master with QUILL_LOGGER_CALL and you pass either nullptr or the tags as one of the arguments.

the similar was done for the LIMIT macros

File: include/quill/LogMacros.h [hunk 2]
```diff
@@ -580,6 +581,51 @@
...
+  #define QUILL_LOGV_NOTICE(logger, message, ...)                                                    \
+    QUILL_LOGGER_CALL(QUILL_LIKELY, logger, quill::LogLevel::Notice,                                 \
+                      QUILL_GENERATE_FORMAT_STRING(message, ##__VA_ARGS__), ##__VA_ARGS__)
+
+  #define QUILL_LOGV_NOTICE_LIMIT(min_interval, logger, message, ...)                                \
+    QUILL_LOGGER_CALL_LIMIT(min_interval, QUILL_LIKELY, logger, quill::LogLevel::Notice,             \
+                            QUILL_GENERATE_FORMAT_STRING(message, ##__VA_ARGS__), ##__VA_ARGS__)
+
+  #define QUILL_LOGV_NOTICE_WITH_TAGS(logger, tags, message, ...)                                    \
+    QUILL_LOGGER_CALL_WITH_TAGS(QUILL_LIKELY, logger, quill::LogLevel::Notice, tags,                 \
```
Comment (odygrd) [inline include/quill/LogMacros.h:603]: This macro has been replaced in master by QUILL_LOGGER_CALL and you pass tags or nullptr if no tags

File: docs/logging_macros.rst [hunk 3]
```diff
@@ -70,6 +70,14 @@ Standard Logging Macros
 
 - :c:macro:`LOG_INFO_TAGS(logger, tags, fmt, ...)`
 
+**Notice**
+
+- :c:macro:`LOG_NOTICE(logger, fmt, ...)`
+
+- :c:macro:`LOG_NOTICE_LIMIT(min_interval, logger, fmt, ...)`
+
+- :c:macro:`LOG_NOTICE_WITH_TAGS(logger, tags, fmt, ...)`
```
Comment (odygrd) [inline docs/logging_macros.rst:79]: Please replace all WITH_TAGS to simply _TAGS to be in line with master

**Review by odygrd — CHANGES_REQUESTED** [review]
Thanks for PR, very appreciated! I am fine adding the extra level it might be useful for some applications

**Comment by Byrdman32** [comment]
Sorry about that, I did all my initial development at v6.1.2 and when I merged up to master I apparently missed some things. 

All the requested changes should be fixed now.

**Review by odygrd — APPROVED** [review]

**Comment by odygrd** [comment]
looks good, thank you : )

**Pull request merged**